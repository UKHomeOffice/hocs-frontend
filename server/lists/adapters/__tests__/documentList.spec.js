const documentListAdapter = require('../documentList');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('Documents List Adapter', () => {

    it('should transform document data and sort by creation date', async () => {
        const mockData = {
            documents: [
                { created: '02-01-2019', displayName: 'Document A', uuid: 1 },
                { created: '01-01-2019', displayName: 'Document B', uuid: 2 },
                { created: '03-01-2019', displayName: 'Document C', uuid: 3 },
            ]
        };

        const results = await documentListAdapter(mockData, { logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});