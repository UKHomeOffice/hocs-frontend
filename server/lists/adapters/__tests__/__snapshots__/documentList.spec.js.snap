// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents List Adapter should transform document data and sort by creation date 1`] = `
[
  {
    "label": "Document C",
    "status": undefined,
    "tags": [
      undefined,
      undefined,
    ],
    "timeStamp": "03-01-2019",
    "value": 3,
  },
  {
    "label": "Document A",
    "status": undefined,
    "tags": [
      undefined,
      undefined,
    ],
    "timeStamp": "02-01-2019",
    "value": 1,
  },
  {
    "label": "Document B",
    "status": undefined,
    "tags": [
      undefined,
      undefined,
    ],
    "timeStamp": "01-01-2019",
    "value": 2,
  },
]
`;
